### PR TITLE
[GEOT-5726] Parse an empty CSSParameter as an empty string instead of null

### PR DESCRIPTION
**Pull request by tbarsballe**
See https://osgeo-org.atlassian.net/browse/GEOT-5726

### CODE REVIEW

File: modules/library/main/src/main/java/org/geotools/styling/SLDParser.java [hunk 1]
```diff
@@ -2308,6 +2308,9 @@ private Expression parseCssParameter(Node root, boolean trimWhiteSpace) {
         for (Expression expression : expressions) {
             ret = manageMixed(ret, expression);
         }
+        if (ret == null) {
+            return ff.literal("");
```
Comment (jodygarnett) [inline modules/library/main/src/main/java/org/geotools/styling/SLDParser.java]: I would prefer to use null for something that is not there?

Comment (jodygarnett) [inline modules/library/main/src/main/java/org/geotools/styling/SLDParser.java]: Double check the GTXML parser handles an empty tag in the same way or we will get into inconsistencies in our model.

File: modules/library/main/src/test/java/org/geotools/styling/SLDTransformerTest.java [hunk 2]
```diff
@@ -834,6 +830,50 @@ public void testTextSymbolizerTransformOutAndInAndOutAgain() throws Exception {
...
+
+        SLDTransformer st = new SLDTransformer();
+        String firstExport = st.transform(style);
+        SLDParser sldp = new SLDParser(CommonFactoryFinder.getStyleFactory(null));
+        sldp.setInput(new StringReader(firstExport));
+        Style[] firstImport = sldp.readXML();
+
+        assertNotNull(firstImport[0]);
+
+        // NPE here
```
Comment (jodygarnett) [inline modules/library/main/src/test/java/org/geotools/styling/SLDTransformerTest.java]: there is no longer a NPE exception here

File: modules/library/main/src/test/java/org/geotools/styling/SLDTransformerTest.java [hunk 3]
```diff
@@ -834,6 +830,50 @@ public void testTextSymbolizerTransformOutAndInAndOutAgain() throws Exception {
         String secondExport = st.transform(firstImport);
 
     }
+
+    @Test
+    public void testTextSymbolizerTransformOutAndInAndOutAndInAgainWithEmptyFontFamily() throws Exception {
```
Comment (jodygarnett) [inline modules/library/main/src/test/java/org/geotools/styling/SLDTransformerTest.java]: for a test case this specific quote the bug number or people will think you are nuts

**Comment by tbarsballe** [comment]
Addressed feedback - added / improved comments.

I added a test case for consistency with the GTXML parser. Turns out it doesn't match the either the current behavior (getting `null`) or the new behavior (getting `""`). Instead, it fills in the default, in this case `"Serif"`. I've added this test case, but set it as ignored, in case we revisit this behavior in the future

**Comment by tbarsballe** [comment]
Backported to 17.x

**Pull request merged**